src/live_restore/live_restore_server.c

```diff
@@ -174,55 +166,72 @@ __live_restore_worker_run(WT_SESSION_IMPL *session, WT_THREAD *ctx)
 }
 
 /*
- * __live_restore_populate_queue --
- *     Populate the live restore work queue. Free all objects on failure.
+ * __insert_queue_item --
+ *     Insert an item into the live restore queue.
  */
 static int
-__live_restore_populate_queue(WT_SESSION_IMPL *session, uint64_t *work_count)
+__insert_queue_item(WT_SESSION_IMPL *session, char *uri, uint64_t *work_count)
 {
     WT_DECL_RET;
-    WT_LIVE_RESTORE_SERVER *server = S2C(session)->live_restore_server;
 
-    TAILQ_INIT(&server->work_queue);
-    /*
-     * Open a metadata cursor to gather the list of objects. The metadata file is built from the
-     * WiredTiger.backup file, during __wt_turtle_init. Thus this function must be run after that
-     * function.
-     */
+    __wt_verbose_debug2(
+      session, WT_VERB_FILEOPS, "Live restore server: Adding %s to the work queue", uri);
 
-    /*
-     * FIXME-WT-13888: Add logic to queue log files first, then the history store. This will use a
-     * directory list call.
-     */
-    WT_CURSOR *cursor;
-    WT_RET(__wt_metadata_cursor(session, &cursor));
     WT_LIVE_RESTORE_WORK_ITEM *work_item = NULL;
+    WT_LIVE_RESTORE_SERVER *server = S2C(session)->live_restore_server;
+
+    WT_ERR(__wt_calloc_one(session, &work_item));
+    WT_ERR(__wt_strdup(session, uri, &work_item->uri));
+    TAILQ_INSERT_HEAD(&server->work_queue, work_item, q);
+    (*work_count)++;
+
+    if (0) {
+err:
+        __wt_free(session, work_item->uri);
+        __wt_free(session, work_item);
+    }
+
+    return (ret);
+}
+
+/*
+ * __live_restore_init_work_queue --
+ *     Populate the live restore work queue. Free all objects on failure.
+ */
+static int
+__live_restore_init_work_queue(WT_SESSION_IMPL *session)
+{
+    WT_DECL_RET;
+    WT_CONNECTION_IMPL *conn = S2C(session);
+    WT_LIVE_RESTORE_SERVER *server = conn->live_restore_server;
+
+    /* Initialize the work queue. */
+    TAILQ_INIT(&server->work_queue);
     __wt_verbose_debug1(
       session, WT_VERB_FILEOPS, "%s", "Live restore server: Initializing the work queue");
 
-    *work_count = 0;
+    WT_CURSOR *cursor;
+    WT_RET(__wt_metadata_cursor(session, &cursor));
+    uint64_t work_count = 0;
     while ((ret = cursor->next(cursor)) == 0) {
-        const char *uri = NULL;
+        char *uri = NULL;
         WT_ERR(cursor->get_key(cursor, &uri));
-        if (WT_PREFIX_MATCH(uri, "file:")) {
-            __wt_verbose_debug2(
-              session, WT_VERB_FILEOPS, "Live restore server: Adding an %s to the work queue", uri);
-            WT_ERR(__wt_calloc_one(session, &work_item));
-            WT_ERR(__wt_strdup(session, uri, &work_item->uri));
-            TAILQ_INSERT_HEAD(&server->work_queue, work_item, q);
-            (*work_count)++;
-        }
+        if (WT_PREFIX_MATCH(uri, "file:"))
+            WT_ERR(__insert_queue_item(session, uri, &work_count));
     }
     WT_ERR_NOTFOUND_OK(ret, false);
 
+    /* Queue the metadata file if we're not restoring from a backup. */
```

Comment:
I'm not familiar with backup, but what I think is happening here is
```suggestion
    /* If we're restoring from a backup we've already run recovery and written the metadata file to the destination. Otherwise it's in source and requires migration. */
```

Reply:
Nah backup recreates the metadata file on startup so it'll be brand new and therefore doesn't need tranfer over. I'll add that detail to my comment.

Reply:
100f331c7